### PR TITLE
codemonitors: Clean up action job store

### PR DESCRIPTION
**Pull request by camdencheek**
This PR cleans up the action job methods on the code monitor
store a bit to simplify the code a bit. Details inline.

Stacked on #27474

### CODE REVIEW

File: enterprise/internal/codemonitors/action_jobs.go [hunk 1]
```diff
@@ -57,32 +55,35 @@ var ActionJobsColumns = []*sqlf.Query{
 }
 
 const readActionEmailEventsFmtStr = `
-SELECT id, email, trigger_event, state, failure_message, started_at, finished_at, process_after, num_resets, num_failures, log_contents
+SELECT %s -- ActionJobsColumns
```
Comment (camdencheek) [inline enterprise/internal/codemonitors/action_jobs.go:58]: Scanning an action job depends on the order of the columns, so make sure we're using the same columns and in the same order for every action job query.

File: enterprise/internal/codemonitors/action_jobs.go [hunk 2]
```diff
@@ -107,105 +108,100 @@ func (s *Store) TotalActionEmailEvents(ctx context.Context, emailID int64, trigg
...
 `
 
 func (s *Store) ActionJobForIDInt(ctx context.Context, recordID int) (*ActionJob, error) {
-	return s.runActionJobQuery(ctx, sqlf.Sprintf(actionJobForIDFmtStr, recordID))
+	q := sqlf.Sprintf(actionJobForIDFmtStr, sqlf.Join(ActionJobsColumns, ", "), recordID)
+	row := s.QueryRow(ctx, q)
+	return scanActionJob(row)
 }
 
-func (s *Store) runActionJobQuery(ctx context.Context, q *sqlf.Query) (ajs *ActionJob, err error) {
```
Comment (camdencheek) [inline enterprise/internal/codemonitors/action_jobs.go:156]: runActionJobQuery was inlined in `ActionJobForIDInt`, and significantly simplified with `QueryRow`.

File: enterprise/internal/codemonitors/action_jobs.go [hunk 3]
```diff
@@ -107,105 +108,100 @@ func (s *Store) TotalActionEmailEvents(ctx context.Context, emailID int64, trigg
...
-	}
-	return ajs, nil
+	return ajs, rows.Err()
+}
+
+type rowScanner interface {
+	Scan(...interface{}) error
+}
+
+func scanActionJob(row rowScanner) (*ActionJob, error) {
```
Comment (camdencheek) [inline enterprise/internal/codemonitors/action_jobs.go:192]: I added `scanActionJob`, which can take either a `*sql.Rows` or a `*sql.Row` to simplify the logic in `scanActionJobs` and in `ActionJobForIDInt`

File: enterprise/internal/codemonitors/action_jobs.go [hunk 4]
```diff
@@ -120,92 +122,86 @@ INSERT INTO cm_action_jobs (email, trigger_event)
 SELECT id, %s::integer from due EXCEPT SELECT id, %s::integer from busy ORDER BY id
 `
 
+// TODO(camdencheek): could we enqueue based on monitor ID rather than query ID? Would avoid joins above.
```
[the remark this answers is not in the document]
Comment (rvantonder) [reply enterprise/internal/codemonitors/action_jobs.go:127]: nit: Maybe this TODO could less open ended like `TODO: enqueuing  monitor ID would avoid joins above` (if we're unsure maybe this is more something to explore first)

**Review by rvantonder — APPROVED** [review]

**Comment by camdencheek** [comment]
I think that might be the best "clean up" gif I've seen yet

**Pull request merged**